Improve text cleaning

AndiPersti opened the pull request:
This PR addresses #2131 and #2138 

In order to improve the text cleaning I've written a new version of `trim()` which removes all 25 Unicode whitespaces by default.

jiru commented:
Thank you for your contribution Andreas, especially the tests.

If I were to implement this, I would have used regexps and Unicode classes, for example:
```php
$frontTrimmed = preg_replace('/^[\p{Zs}\p{Cc}]/u', '', $sentence);
$endTrimmed = preg_replace('/[\p{Zs}\p{Cc}]$/u', '', $sentence);
$middleReplaced = preg_replace('/[\p{Zs}\p{Cc}]{2,}/u', ' ', $sentence);
```
`\p{Cc}` represents [all control characters](https://www.compart.com/en/unicode/category/Cc) and `\p{Zs}` [all spaces](https://www.compart.com/en/unicode/category/Zs).

What do you think?

AndiPersti commented:
> If I were to implement this, I would have used regexps and Unicode classes, for example:

Actually I've thought about using regexps but I can't remember now why I decided not to use them. But they are definitely a better solution.

I've updated the PR with the following changes:
- Used `\p{Z}` because we also want to remove `LINE SEPARATOR` (category `Zl`) and `PARAGRAPH SEPARATOR` (category `Zp`).
- Used `+` quantifier to catch more than one problematic character and added a test for this case.
- Modified the regex for reducing consecutive whitespace characters to one as suggested.
- Removed `unicode_trim()` and the tests for it

jiru commented:
Looks good, thank you! :+1:
